src/builder/command.rs

```diff
@@ -274,6 +274,51 @@ impl<'help> App<'help> {
         self
     }
 
+    /// Allows one to mutate a [`Command`] after it's been added as a subcommand.
+    ///
+    /// This can be useful for modifying auto-generated arguments of nested subcommands with
+    /// [`Command::mut_arg`].
+    ///
+    /// # Examples
+    ///
+    /// ```rust
+    /// # use clap::Command;
+    ///
+    /// let mut cmd = Command::new("foo")
+    ///         .subcommand(Command::new("bar"))
+    ///         .mut_subcommand("bar", |subcmd| subcmd.disable_help_flag(true));
+    ///
+    /// let res = cmd.try_get_matches_from_mut(vec!["foo", "bar", "--help"]);
+    ///
+    /// // Since we disabled the help flag on the "bar" subcommand, this should err.
+    ///
+    /// assert!(res.is_err());
+    ///
+    /// let res = cmd.try_get_matches_from_mut(vec!["foo", "bar"]);
+    /// assert!(res.is_ok());
+    /// ```
+    #[must_use]
+    pub fn mut_subcommand<T, F>(mut self, name: T, f: F) -> Self
+    where
+        F: FnOnce(App<'help>) -> App<'help>,
+        T: Into<&'help str>,
+    {
+        let subcmd_name: &str = name.into();
+        let pos = self
+            .subcommands
+            .iter()
+            .position(|s| s.aliases_to(subcmd_name));
```

Comment:
imo we should only `mut` using the literal name and not an alias.  That feels a bit too sloppy for an API design (iirc no other way of referring to a subcommand or arg works with aliases)

Reply:
makes sense, I pushed a change to not match on subcommand aliases and mirror `mut_arg`'s logic.

one issue I could see from not matching on aliases though is if say
1. Suppose there's an existing subcommand with name `foo` and alias `foobar`.
2. Call `mut_subcommand("foobar", |cmd| ...)`, since it doesn't find a subcommand with an ID that matches "foobar", this inserts a new subcommand with name "foobar"
3. There's now a subcommand that has an alias for "foobar" and a subcommand with the actual name "foobar"

this would result in the "foobar" command being completely ignored right as, iirc, clap will use the first matching subcommand? this might result in confusing behavior, and also curious if this is a problem with `mut_arg`?

Reply:
> this would result in the "foobar" command being completely ignored right as, iirc, clap will use the first matching subcommand? this might be confusing for users of Clap as it would appear that mut_subcommand took no effect in this situation. also curious if this is a problem with mut_arg?

We shouldn't allow people to do that; it is an oversight.  See #3888.